dashboard/lib/delayed_job_manager.rb

```diff
@@ -0,0 +1,46 @@
+require 'cdo/aws/metrics'
+
+module DelayedJobManager
+  METRICS_NAMESPACE = 'code-dot-org/ActiveJob'.freeze
+
+  # Delayed::Job performance degrades sharply once the main `delayed_jobs` table
+  # accumulates too many rows, which often happens when failures pile up. We
+  # archive those failures into a secondary table so engineers can still inspect
+  # them while keeping the primary queue lean. Long term we plan to migrate
+  # queue processing to SolidQueue, which is blocked until we're on Rails 7.
+  def archive_failed_jobs
+    return unless FailedDelayedJob.table_exists?
+
+    archived_count = 0
+    Delayed::Job.where.not(failed_at: nil).find_in_batches do |batch|
+      ActiveRecord::Base.transaction do
+        failed_jobs_attributes = batch.map(&:attributes)
+        FailedDelayedJob.insert_all(failed_jobs_attributes)
+        Delayed::Job.where(id: batch.map(&:id)).delete_all
+      end
+      archived_count += batch.size
+    end
+
+    record_metric('FailedJobsArchived', archived_count)
+    archived_count
+  rescue
+    record_metric('FailedJobArchiveFailures', 1)
+    raise
+  end
+
+  def record_metric(metric_name, value)
```

Comment:
This might be getting a bit too into the weeds, but what would you think about defining this method in `ActiveJobMetrics` instead, and invoking it from here? I'm not sure how I feel about entangling these two modules, but it does feel like an appropriate division of responsibilities